staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1040,12 +1040,22 @@ func (c *cacheWatcher) process(initEvents []*watchCacheEvent, resourceVersion ui
 		klog.V(2).Infof("processing %d initEvents of %s took %v", len(initEvents), objType, processingTime)
 	}
 
+	timer := time.NewTimer(timeout)
```

Comment:
Creating a Timer allocates memory, which we would like to avoid. I think we can avoid that if:
- we create a sync.Pool of Timers (we used to have sth like that previously - you can take a look into https://github.com/kubernetes/kubernetes/pull/73846/files that removed it)
- we will add a `timer` field to cacheWatcher and will be initializing it from that pool at cacheWatcher creation